main/gridinit.c

```diff
@@ -321,6 +321,105 @@ command_stop(GString *out, int argc, char **argv)
 	return service_run_groupv(argc, argv, out, stop_process);
 }
 
+static void
+command_signal(GString *out, int argc, char **argv)
+{
+	char *str_signal;
+	int signal;
+
+	static struct signal_mapping_s {
+		char *name;
+		int value;
+	} SIGNAL [] = {
+#ifdef __unix__ // the base signals
+		{ "SIGABRT",	SIGABRT},
+		{ "SIGALRM",	SIGALRM },
+		{ "SIGBUS",	SIGBUS },
+		{ "SIGCHLD",	SIGCHLD },
+		{ "SIGCONT",	SIGCONT },
+		{ "SIGEMT",	SIGEMT },
+		{ "SIGFPE",	SIGFPE },
+		{ "SIGHUP",	SIGHUP },
+		{ "SIGILL",	SIGILL },
+		{ "SIGINFO",	SIGINFO },
+		{ "SIGINT",	SIGINT },
+		{ "SIGIO",	SIGIO },
+		{ "SIGKILL",	SIGKILL },
+		{ "SIGPIPE",	SIGPIPE },
+		{ "SIGPROF",	SIGPROF },
+		{ "SIGQUIT",	SIGQUIT },
+		{ "SIGSEGV",	SIGSEGV },
+		{ "SIGSTOP",	SIGSTOP },
+		{ "SIGSYS",	SIGSYS },
+		{ "SIGTERM",	SIGTERM },
+		{ "SIGTRAP",	SIGTRAP },
+		{ "SIGTSTP",	SIGTSTP },
+		{ "SIGTTIN",	SIGTTIN },
+		{ "SIGTTOU",	SIGTTOU },
+		{ "SIGURG",	SIGURG },
+		{ "SIGUSR1",	SIGUSR1 },
+		{ "SIGUSR2",	SIGUSR2 },
+		{ "SIGVTALRM",	SIGVTALRM },
+		{ "SIGWINCH",	SIGWINCH },
+		{ "SIGXCPU",	SIGXCPU },
+		{ "SIGXFSZ",	SIGXFSZ },
+#else
+#error Your plateform is not supported
+#endif
+
+#ifdef __FreeBSD__ // freebsd specific signals
+		{ "SIGTHR",	SIGTHR },
+		{ "SIGLIBRT",	SIGLIBRT },
+#endif
+#ifdef __linux__ // linux specific signals
+		{ "SIGCLD",	SIGCLD },
+		{ "SIGIOT",	SIGIOT },
+		{ "SIGLOST",	SIGLOST },
+		{ "SIGPOLL",	SIGPOLL },
+		{ "SIGPWR",	SIGPWR },
+		{ "SIGSTKFLT",	SIGSTKFLT },
+		{ "SIGUNUSED",	SIGUNUSED },
+		{ NULL, 0 },
+#endif
```

Comment:
I don’t think we should put this thing here. It’s not good for code readability.
It would probably be best to create a `plateform` file or a `signal` file for this.

Reply:
:+1: to have a dedicated file